src/w3d/renderer/w3d_util.cpp

```diff
@@ -0,0 +1,73 @@
+/**
+ * @file
+ *
+ * @author Jonathan Wilson
+ *
+ * @brief W3D Utillity class
+ *
+ * @copyright Thyme is free software: you can redistribute it and/or
+ *            modify it under the terms of the GNU General Public License
+ *            as published by the Free Software Foundation, either version
+ *            2 of the License, or (at your option) any later version.
+ *            A full copy of the GNU General Public License can be found in
+ *            LICENSE
+ */
+#include "w3d_util.h"
+#include "shader.h"
+#include "vector3.h"
+#include "vector4.h"
+#include "w3d_file.h"
+
+void W3dUtilityClass::Convert_Vector(const IOVector3Struct &v, Vector3 *set)
+{
+    set->X = v.x;
+    set->Y = v.y;
+    set->Z = v.z;
+}
+
+void W3dUtilityClass::Convert_Color(const W3dRGBStruct &rgb, Vector3 *set)
+{
+    set->X = (float)rgb.r / 255;
```

Comment:
I saw this before and wondered why the `(float)`? You are dividing by a integer and storing in a float.

Reply:
The float conversion ensures the / operator does a float divide and not an int divide

Reply:
then why not change the constant to `255.0f`